raiden-ts/src/transfers/epics/withdraw.ts

```diff
@@ -1,78 +1,370 @@
+            const processed: Processed = {
+              type: MessageType.PROCESSED,
+              message_identifier: confirmation.message_identifier,
+            };
+            processed$ = from(signMessage(signer, processed, { log })).pipe(
+              tap((signed) => cache.put(cacheKey, signed)),
+            );
+          }
+
+          // don't validate req here, to always update nonce, but do on tx send
+          return processed$.pipe(
+            map((processed) =>
+              messageSend.request(
+                { message: processed },
+                { address: action.meta.partner, msgId: processed.message_identifier.toString() },
+              ),
+            ),
+          );
+        }),
+        catchError(
+          (err) => (
+            log.info('Signing Processed message for WithdrawConfirmation failed, ignoring', err),
+            EMPTY
+          ),
+        ),
+      ),
+    ),
+  );
+};
 
 /**
- * Remove [[WithdrawRequest]] from channel end's pending withdraw requests when they expire
+ * Dispatch withdrawExpire.request when one of our sent WithdrawRequests expired
  *
  * @param action$ - Observable of newBlock actions
  * @param state$ - Observable of RaidenStates
  * @param deps - Epics dependencies
  * @param deps.config$ - Config observable
  * @returns Observable of withdrawExpired actions
  */
-export const autoExpireWithdrawEpic = (
+export const autoWithdrawExpireEpic = (
   action$: Observable<RaidenAction>,
   state$: Observable<RaidenState>,
   { config$ }: RaidenEpicDeps,
-): Observable<withdrawExpired> =>
+): Observable<withdrawExpire.request> =>
   action$.pipe(
     filter(newBlock.is),
     pluck('payload', 'blockNumber'),
     withLatestFrom(state$, config$),
     mergeMap(function* ([blockNumber, state, { confirmationBlocks }]) {
       for (const channel of Object.values(state.channels)) {
         if (channel.state !== ChannelState.open) continue;
-        for (const [end, mul] of confMult) {
-          for (const request of channel[end].withdrawRequests) {
-            // mul here makes us wait double confirmation before expiring our 'own' requests
-            // to give partner time to catch up with block and expire them
-            if (request.expiration.add(confirmationBlocks * mul).gt(blockNumber)) continue;
-            yield withdrawExpired(
-              { participant: channel[end].address },
-              {
-                tokenNetwork: channel.tokenNetwork,
-                partner: channel.partner.address,
-                expiration: request.expiration.toNumber(),
-                totalWithdraw: request.total_withdraw,
-              },
-            );
-          }
+        for (const req of channel.own.pendingWithdraws) {
+          if (req.type !== MessageType.WITHDRAW_REQUEST) continue;
+          if (req.expiration.add(confirmationBlocks).gt(blockNumber)) continue;
+          // if there's already an WithdrawExpired in 'pendingWithdraws' array, don't request again
```

Comment:
Comment doesn't add anything the code isn't already expressing itself without having anything implicit thing that is hidden.